src/components/data/data-utils.js

```diff
@@ -361,19 +353,29 @@ export const EntityUploader = ({ onSuccess, onDismiss, namespace, name, entityTy
         title: 'Import Table Data',
         width: '35rem',
         okButton: h(ButtonPrimary, {
-          disabled: !currentFile || isInvalid || uploading,
-          tooltip: !currentFile || isInvalid ? 'Please select valid data to upload' : 'Upload selected data',
+          disabled: dataProvider.tsvFeatures.disabled(currentFile, isInvalid, uploading, recordType),
+          tooltip: dataProvider.tsvFeatures.tooltip(currentFile, isInvalid, recordType),
           onClick: doUpload
         }, ['Start Import Job'])
       }, [
         div({ style: { padding: '0 0 1rem' } },
-          ['Choose the data import option below. ',
+          [dataProvider.tsvFeatures.uploadInstructions,
             h(Link, {
               ...Utils.newTabLinkProps,
               href: 'https://support.terra.bio/hc/en-us/articles/360025758392'
             }, ['Click here for more info on the table.']),
             p(['Data will be saved in location: 🇺🇸 ', span({ style: { fontWeight: 'bold' } }, 'US '), '(Terra-managed).'])]),
-        h(SimpleTabBar, {
+        dataProvider.tsvFeatures.needsTypeInput && div([div({ style: { padding: '0 0 1rem' } }, 'Table name:  '),
+          h(TextInput, {
+            id: 'recordTypeInput',
+            value: recordType,
+            placeholder: 'Enter table name',
+            onChange: value => {
+              setRecordType(value)
+              setRecordAlreadyExists(_.includes(value, entityTypes))
+            }
+          })]),
+        dataProvider.features.supportsTabBar && h(SimpleTabBar, {
```

Comment:
Is there any difference between file import and text import? If a provider supports TSV upload, shouldn't it support both?